extensions/notebook/src/test/book/book.test.ts

```diff
@@ -73,26 +73,30 @@ describe('BookTreeViewProvider.getChildren', function (): void {
 	});
 
 	it('should return all book nodes when element is undefined', async function (): Promise<void> {
-		const children = await bookTreeViewProvider.getChildren();
-		should(children).be.Array();
-		should(children.length).equal(1);
-		book = children[0];
-		should(book.title).equal(expectedBook.title);
+		bookTreeViewProvider.onReadAllTOCFiles(async () => {
```

Comment:
This could hang forever if it fails. I guess the test will timeout in that case? Would be good to have some kind of logic that wait on event or timeout, whichever happens 1st and then throws if timeout fails. E.g.
```ts
let tocRead = new Promise((resolve, reject) => bookTreeViewProvider.onReadAllTOCFiles(() => resolve()));
let errorCase = timeout(150).then(() => throw new Error('TOC was not ready in time'));
await Promise.race([tocRead, errorCase)]);
// Then work on the book
```

Not sure if this exact code will work, but something similar should.

Reply:
Merging for now but I will add the timeout logic in another PR. Thank you!